src/pokenav_region_map.c

```diff
@@ -204,17 +208,22 @@ u32 GetRegionMapCallback(void)
 
 static u32 HandleRegionMapInput(struct Pokenav_RegionMapMenu *state)
 {
+    struct RegionMap* regionMap = GetSubstructPtr(POKENAV_SUBSTRUCT_REGION_MAP);
+
     switch (DoRegionMapInputCallback())
     {
-    case MAP_INPUT_MOVE_END:
-        return POKENAV_MAP_FUNC_CURSOR_MOVED;
-    case MAP_INPUT_A_BUTTON:
-        if (!IsRegionMapZoomed())
-            return POKENAV_MAP_FUNC_ZOOM_IN;
-        return POKENAV_MAP_FUNC_ZOOM_OUT;
-    case MAP_INPUT_B_BUTTON:
-        state->callback = GetExitRegionMapMenuId;
-        return POKENAV_MAP_FUNC_EXIT;
+        case MAP_INPUT_MOVE_END:
+            return POKENAV_MAP_FUNC_CURSOR_MOVED;
+        case MAP_INPUT_A_BUTTON:
+            if (!IsRegionMapZoomed())
+                return POKENAV_MAP_FUNC_ZOOM_IN;
+            return POKENAV_MAP_FUNC_ZOOM_OUT;
+        case MAP_INPUT_B_BUTTON:
+            state->callback = GetExitRegionMapMenuId;
+            return POKENAV_MAP_FUNC_EXIT;
+        case MAP_INPUT_R_BUTTON:
+            if (regionMap->mapSecType == MAPSECTYPE_CITY_CANFLY && OW_FLAG_USE_FLY_FROM_POKENAV)
+                return POKENAV_MAP_FUNC_FLY;
```

Comment:
we don't usually indent case blocks like that so it was fine as is.

Reply:
i don't remember changing that, but i'll dedent it!

Reply:
fixed!